tools/ci_credentials/ci_credentials/secrets_loader.py

```diff
@@ -0,0 +1,240 @@
+import base64
+import json
+import re
+from dataclasses import dataclass
+from json.decoder import JSONDecodeError
+from pathlib import Path
+from typing import Mapping, Any, Tuple, ClassVar
+
+from ci_common_utils import GoogleApi
+from ci_common_utils import Logger
+
+DEFAULT_SECRET_FILE = "config"
+DEFAULT_SECRET_FILE_WITH_EXT = DEFAULT_SECRET_FILE + ".json"
+
+GSM_SCOPES = ("https://www.googleapis.com/auth/cloud-platform",)
+
+
+@dataclass
```

Comment:
This is debatable but I prefer the use of `@dataclass` specifically for more storage-oriented classes, replacing the usage of NamedTuples. 

In this case I think you're only using the __init__, so I would argue for clarity this doesn't need to be a dataclass. Wdyt?

Reply:
ok. changed